Source/WebCore/rendering/StyledMarkedText.cpp

```diff
@@ -136,11 +143,42 @@ Vector<StyledMarkedText> StyledMarkedText::subdivideAndResolve(const Vector<Mark
         return { styledMarkedText };
     }
 
-    auto markedTexts = MarkedText::subdivide(textsToSubdivide);
+    auto markedTexts = MarkedText::subdivide(textsToSubdivide, OverlapStrategy::None);
     ASSERT(!markedTexts.isEmpty());
     if (UNLIKELY(markedTexts.isEmpty()))
         return { };
 
+    if (!markedTexts.isEmpty()) {
+        // Get rid of duplicates in vector
+        ListHashSet<MarkedText> noDuplicates;
+        for (auto& elem : markedTexts)
+            noDuplicates.add(elem);
+        // Add back to markedTexts
+        if (!noDuplicates.isEmpty()) {
+            markedTexts.clear();
+            for (auto& elem : noDuplicates)
+                markedTexts.append(elem);
+        }
+        // Sort so highlights follow correct priority/insertion order
+        if (!markedTexts.first().highlightName.isNull()) {
+            // Maintain copy of old unordered markedTexts
+            Vector<MarkedText> copyOfMarkedTexts;
+            copyOfMarkedTexts = markedTexts;
```

Comment:
This can be a single assignment.
Also, instead of naming this `copyOfMarkedTexts`, consider `unorderedMarkedTexts`?